server/controllers/users/users.ts

```diff
@@ -1,236 +1,306 @@
-		if (user) {
-			const { firstName, lastName, password, emailVerified, roles } = req.body;
-			let { email } = req.body;
-
-			if (firstName) user.firstName = firstName;
-			if (lastName) user.lastName = lastName;
-
-			if(req.user.admin){
-				if (emailVerified) user.emailVerified = emailVerified;
-				if (roles) user.roles = roles;
-			}
-
-			if (email) {
-				email = email.toLowerCase();
-				if (email !== user.email) {
-					const emailExist = await User.findOne({ email });
-					if (emailExist) {
-						return res
-							.status(400)
-							.json({ errors: [{ msg: messages.EMAIL_ALREADY_EXISTS }] });
-					}
-					user.email = email;
-					if(!req.user.admin){
-						user.activationKey = bcrypt.hashSync(email).replace(/\//g, '');
-						user.emailVerified = false;
-						sendVerificationEmail(user);
-					}
-				}
-			}
-			if (password) {
-				user.password = bcrypt.hashSync(password);
-			}
-			await user.save();
-			return res.json({ msg: messages.USER_UPDATED });
-		}
-		return res.json({ msg: messages.USER_NOT_FOUND });
-	} catch {
-		return res.status(500).send(messages.SERVER_ERROR);
-	}
-};
+  try {
+    const user = await User.findById(req.user.id);
+    if (user) {
+      const { firstName, lastName, password, emailVerified, roles } = req.body;
+      let { email } = req.body;
 
-interface ITransporterAuth {
-	user: string;
-	pass: string;
-}
-const sendVerificationEmail = (user: IUser) => {
-	const mailSettings: TransportOptions = config.get('mail.settings');
-	const mailCredentials: ITransporterAuth = { 
-		user: process.env.MAIL_USER || config.get('mail.credentials.user'), 
-		pass: process.env.MAIL_PASS || config.get('mail.credentials.pass') 
-	}
-	const transporter = nodemailer.createTransport({
-		...mailSettings,
-		auth: mailCredentials
-	});
-
-	const mailOptions = {
-		from: 'support <natripareact@gmail.com>',
-		to: user.email,
-		subject: 'Account Activation',
-		html: `
-      To activate your account, click this link
-      <a href="http://localhost:5000/api/users/activate/${
-				user.activationKey
-			}">Activate Account</a>
-    `
-	};
-
-	transporter.sendMail(mailOptions, (err, data) => {
-		if (err) throw err;
-	});
+      if (firstName) user.firstName = firstName;
+      if (lastName) user.lastName = lastName;
+
+      if (req.user.admin) {
+        if (emailVerified) user.emailVerified = emailVerified;
+        if (roles) user.roles = roles;
+      }
+
+      if (email) {
+        email = email.toLowerCase();
+        if (email !== user.email) {
+          const emailExist = await User.findOne({ email });
+          if (emailExist) {
+            return res
+              .status(400)
+              .json({ errors: [{ msg: messages.EMAIL_ALREADY_EXISTS }] });
+          }
+          user.email = email;
+          if (!req.user.admin) {
+            user.activationKey = bcrypt.hashSync(email).replace(/\//g, "");
+            user.emailVerified = false;
+            sendVerificationEmail(user);
```

Comment:
It won't work, this function was extracted to another file

Reply:
fixed